### PR TITLE
Msal-browser event API documentation

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR adds docs for events in msal-browser.

### CODE REVIEW

File: lib/msal-browser/docs/events.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Events
+
+MSAL.js 2.0 provides event APIs that are available to users of our core library and wrapper libraries. These events are related to auth and what MSAL is doing, and can be used in applications to update UI, show error messages, and so on.
```
Comment (DarylThayil) [inline lib/msal-browser/docs/events.md:3]: 2.x? or maybe Msal Browser since we may have other major versions of this

Comment (jasonnutter) [reply lib/msal-browser/docs/events.md:3]: "MSAL.js v2"  maybe?

File: lib/msal-browser/docs/events.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+## How to use the Event API
+Msal-browser exports the `addEventCallback` function which takes in a callback function and can be used to process emitted events. Events are emitted in msal-browser according to the list below.
+
+Here is an example of how you could consume the emitted events in your application:
+```javascript
+msalInstance.addEventCallback((message: EventMessage) => {
+    // Update UI or interact with EventMessage here
+    if (message.eventType === EventType.LOGIN_SUCCESS) {
+        console.log(message.payload);
+});
```
Comment (jasonnutter) [inline lib/msal-browser/docs/events.md:25]:
```suggestion
msalInstance.addEventCallback((message: EventMessage) => {
    // Update UI or interact with EventMessage here
    if (message.eventType === EventType.LOGIN_SUCCESS) {
        console.log(message.payload);
     }
});
```

File: lib/msal-browser/docs/events.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+## List of Event Types
+
+This the list of events currently emitted by msal-browser.
+
+| Event Type  | Description  | Interaction Type  | Payload   | Error |
+|------------ |:----------: |:-----------: |:------------: | :---------:|
+| `LOGIN_START` | LoginPopup or loginRedirect are called | `Popup` or `Redirect` | PopupRequest or RedirectRequest |
```
Comment (jasonnutter) [inline lib/msal-browser/docs/events.md:34]: For readability during editing, its often nice to line up the columns.

File: lib/msal-browser/docs/events.md (hunk 1 above)
Comment (tnorling) [reply lib/msal-browser/docs/events.md:3]: 2.4+?

File: lib/msal-browser/docs/events.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+| `ACQUIRE_TOKEN_FAILURE` | Error when acquiring token |`Popup` or `Redirect` or `Silent`|  | AuthError |
+| `ACQUIRE_TOKEN_NETWORK_START` | Starting acquiring token from network |`Silent`|               |            
+| `SSO_SILENT_START` | SsoSilent API called |`Silent`|SsoSilentRequest|
+| `SSO_SILENT_SUCCESS` | SsoSilent succeeded |`Silent`| AuthenticationResult |
+| `SSO_SILENT_FAILURE` | SsoSilent failed |`Silent`|  | AuthError |
+| `HANDLE_REDIRECT_START` | HandleRedirectPromise called and hash being processed | `Redirect` | | 
+| `LOGOUT_START` | Logout called |`Redirect`| EndSessionRequest |
+| `LOGOUT_SUCCESS` | Logout success |`Redirect`|EndSessionRequest|
+| `LOGOUT_FAILURE`| Logout failed|`Redirect`|  | Error |
+|
```
Comment (hectormmg) [inline lib/msal-browser/docs/events.md:48]:
```suggestion

```

**Review by jasonnutter — APPROVED** [review]
A couple small things but otherwise looks good!

**Review by hectormmg — APPROVED** [review]
Just noticed a rogue pipe character at the end, otherwise looks good to me!

File: lib/msal-browser/docs/events.md (hunk 1 above)
Comment (hectormmg) [reply lib/msal-browser/docs/events.md:3]: I'd vote for "MSAL Browser starting at version 2.4"

Comment (pkanher617) [reply lib/msal-browser/docs/events.md:3]: don't mean to add to the chaos here, but I would go with package name `@azure/msal-browser`. I think it's the easiest way to distinguish between the different code bases we maintain.

File: lib/msal-browser/docs/events.md (hunk 3 above)
Comment (jo-arroyo) [reply lib/msal-browser/docs/events.md]: Makes sense. Done.

File: lib/msal-browser/docs/events.md (hunk 1 above)
Comment (jo-arroyo) [reply lib/msal-browser/docs/events.md]: Thanks all for the suggestions! I have changed it to incorporate mostly everything. Let me know if you strongly disagree with the change/have more suggestions.

File: lib/msal-browser/docs/events.md [hunk 5]
```diff
@@ -30,19 +30,19 @@ msalInstance.addEventCallback((message: EventMessage) => {
...
+|:-----------------------------:|:--------------------------------------------------------------------------:|:--------------------------------:|:-----------------------------------------------:|:------------------:|
+| `LOGIN_START`                 | LoginPopup or loginRedirect are called                                     | `Popup` or `Redirect`            | PopupRequest or RedirectRequest                 |                    |
+| `LOGIN_SUCCESS`               | Successfully logged in                                                     | `Popup` or `Redirect`            | AuthenticationResult                            |                    |
+| `LOGIN_FAILURE`               | Error when logging in                                                      | `Popup` or `Redirect`            |                                                 | AuthError or Error |
+| `ACQUIRE_TOKEN_START`         | AcquireTokenPopup or acquireTokenRedirect or acquireTokenSilent are called | `Popup` or `Redirect` or `Silent`| PopupRequest or RedirectRequest or SilentRequest|                    |
```
Comment (sameerag) [inline lib/msal-browser/docs/events.md:43]: may be `is` called?

**Review by sameerag — APPROVED** [review]
Approved. Do we want sample code on how to use these events?

File: lib/msal-browser/docs/events.md (hunk 5 above)
Comment (jo-arroyo) [reply lib/msal-browser/docs/events.md]: Good catch! Changed.

**Pull request merged**